crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +375,88 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    pub async fn save_settlement(
+        &self,
+        event: domain::eth::Event,
+        auction: domain::auction::Id,
+        settlement: Option<&domain::settlement::Settlement>,
+    ) -> Result<(), error::Settlement> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["save_settlement"])
+            .start_timer();
+
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .map_err(error::Settlement::BadCommunication)?;
```

Comment:
nit: I still think BadCommunication is a badly communicated name for a Persistence/Db error.